examples/src/main/scala/sttp/tapir/examples/observability/OpenTelemetryMetricsExample.scala

```diff
@@ -0,0 +1,98 @@
+  * {{{
+  *   docker run -p 4317:4317 otel/opentelemetry-collector:latest
+  * }}}
+  *
+  * Please refer to the <a href="https://opentelemetry.io/docs/concepts/sdk-configuration/otlp-exporter-configuration/">exporter
+  * configuration</a> if you need to use a different host/port.
+  *
+  * The example code requires the following dependencies:
+  * {{{
+  *   val openTelemetryVersion = <OpenTelemetry version>
+  *
+  *   libraryDependencies ++= Seq(
+  *     "io.opentelemetry" % "opentelemetry-sdk" % openTelemetryVersion,
+  *     "io.opentelemetry" % "opentelemetry-sdk-metrics" % openTelemetryVersion,
+  *     "io.opentelemetry" % "opentelemetry-exporter-otlp" % openTelemetryVersion
+  *    )
+  * }}}
+  *
+  * Once this example app and the collector are running, and after you send some requests to the `/person` endpoint, you should start seeing
+  * the metrics in the collector logs (look for `InstrumentationScope tapir 1.0.0`), e.g.:
+  * {{{
+  *   InstrumentationScope tapir 1.0.0
+  *   Metric #0
+  *   Descriptor:
+  *        -> Name: request_active
+  *        -> Description: Active HTTP requests
+  *        -> Unit: 1
+  *        -> DataType: Sum
+  *        -> IsMonotonic: false
+  *        -> AggregationTemporality: Cumulative
+  *
+  *   ...
+  * }}}
+  */
+object OpenTelemetryMetricsExample extends App with StrictLogging {
+
+  case class Person(name: String)
+
+  // Simple endpoint returning 200 or 400 response with string body
+  val personEndpoint: ServerEndpoint[Any, Future] =
+    endpoint.post
+      .in("person")
+      .in(jsonBody[Person])
+      .out(stringBody)
+      .errorOut(stringBody)
+      .serverLogic { p =>
+        Thread.sleep(3000)
+        Future.successful(Either.cond(p.name == "Jacob", "Welcome", "Unauthorized"))
+      }
+
+  // An exporter that sends metrics to a collector over gRPC
+  val grpcExporter = OtlpGrpcMetricExporter.builder().build()
+
+  // A metric reader that exports using the gRPC exporter
+  val metricReader: PeriodicMetricReader = PeriodicMetricReader.builder(grpcExporter).build()
+
+  // A meter registry whose meters are read by the above reader
+  val meterProvider: SdkMeterProvider = SdkMeterProvider.builder().registerMetricReader(metricReader).build()
+
+  // An instance of OpenTelemetry using the above meter registry
+  val otel: OpenTelemetry = OpenTelemetrySdk.builder().setMeterProvider(meterProvider).build()
+
+  val openTelemetryMetrics = OpenTelemetryMetrics.default[Future](otel)
+
+  val serverOptions: NettyFutureServerOptions =
+    NettyFutureServerOptions.customiseInterceptors
+      // Adds an interceptor which collects metrics by executing callbacks
+      .metricsInterceptor(openTelemetryMetrics.metricsInterceptor())
+      .options
+
+  Await.ready(NettyFutureServer().port(8080).addEndpoint(personEndpoint, serverOptions).start(), 1.minute)
+
+  logger.info(s"Server started. POST persons under http://localhost:8080/person.")
```

Comment:
maybe also add an example curl command? writing the json by hand could be tedious :)

Reply:
Here you go :)